lib/hotwalletUtils.js

```diff
@@ -117,12 +117,6 @@ exports.waitForNewTransaction = async function waitForNewTransaction(envs, hwRed
     }
   }
 
-  const hasTokens = await blockchain.positiveTokenBalance(hwAddress)
```

Comment:
1) Does this have any implications during wallet initialization when the wallet has no balance?
2) Also, when the wallet doesn't have tokens it shouldn't try to perform ordinary token transfer transactions. Checking that removing this check doesn't interfere with that case either.

If it doesn't affect these 2 cases can you add links to where the tests for this are or add tests for these cases?

Thanks!

Reply:
> Does this have any implications during wallet initialization when the wallet has no balance?

Nope, wallet doesn't need to have any token balance to perform the initialistion and registration, only ETH to pay approval tx fee or ALGO to pay opt-in fee.

> when the wallet doesn't have tokens it shouldn't try to perform ordinary token transfer transactions. 

This is not possible currently, since to receive the withdraw command wallet has to reach the API first.
If we don't allow the wallet reach the API with zero token balance, we would have the same problem when withdrawals are not possible until we top up the wallet with tokens.

As I mentioned above there's nothing wrong with allowing wallet to pickup a transaction with zero balance, since we are still checking the balance (twice actually, since web3 also checks that) to make sure that it's enough for the transfers, and if it's not enough (or zero) user would get a meaningful error message on CoMakery and transfer would immediately be returned to queue.

For the spec pls look at:
https://github.com/CoMakery/robowallet/blob/6e5713e5f1d6cf116d2f1ec76b50a2ddb3f44ec1/tests/waitForNewTransaction.test.js#L39